Introduce snow/desert-coverage and custom terrain type, and move "maximum map height" to settings only (#8891)
OpenTTD/OpenTTD: src/lang/english.txt

STR_TERRAIN_TYPE_HILLY                                          :Hilly
STR_TERRAIN_TYPE_MOUNTAINOUS                                    :Mountainous
STR_TERRAIN_TYPE_ALPINIST                                       :Alpinist
STR_TERRAIN_TYPE_CUSTOM                                         :Custom

> STR_TERRAIN_TYPE_CUSTOM_VALUE                                   :Custom ({NUM})

Reviewer: `Custom height`? `Custom peak height` ? might be a bit long, but just "Custom (Number)" feels unclear as to what it means

Author: I tried so many combinations by now .. without also renaming "Terrain Type", it seems impossible to get a good combination. Going with `Custom height` for now. Better than `Custom`, but not really accurate. Can't have them all :D

STR_CONFIG_SETTING_CITY_APPROVAL                                :Town council's attitude towards area restructuring: {STRING2}
STR_CONFIG_SETTING_CITY_APPROVAL_HELPTEXT                       :Choose how much noise and environmental damage by companies affect their town rating and further construction actions in their area

STR_CONFIG_SETTING_MAP_HEIGHT_LIMIT                             :Map height limit: {STRING2}

> STR_CONFIG_SETTING_MAP_HEIGHT_LIMIT_HELPTEXT                    :Set the maximum allowed height of the map; neither the terrain generator nor you can build higher than this limit. On (auto) it will pick a good value after terrain generation

Reviewer: Second sentence feels unnecessary to me
```suggestion
STR_CONFIG_SETTING_MAP_HEIGHT_LIMIT_HELPTEXT                    :Set the maximum height of the map terrain. With (auto) a good value will be picked after terrain generation
```
Can we put (auto) in quotes as well? to more clearly denote it as an option

STR_CONFIG_SETTING_MAP_HEIGHT_LIMIT_VALUE                       :{NUM}
STR_CONFIG_SETTING_MAP_HEIGHT_LIMIT_AUTO                        :(auto)

> STR_CONFIG_SETTING_TOO_HIGH_MOUNTAIN                            :{WHITE}You can't set the map height limit to this value. At least one mountain on the map is higher

Reviewer: "on the current map" ?

Author: Lol, that part of the string has been there since 2004 :P

Author: A second look, not sure what "current" adds to the sentence. It gives the suggestion there is another map, but for the game you are playing, there isn't. Might be a bit nitpicking back, or bikeshedding, but not sure that adding "current" is a good idea here.

STR_CONFIG_SETTING_INDUSTRY_DENSITY_HELPTEXT                    :Set how many industries should be generated and what level should be maintained during the game
STR_CONFIG_SETTING_OIL_REF_EDGE_DISTANCE                        :Maximum distance from edge for Oil industries: {STRING2}
STR_CONFIG_SETTING_OIL_REF_EDGE_DISTANCE_HELPTEXT               :Limit for how far from the map border oil refineries and oil rigs can be constructed. On island maps this ensures they are near the coast. On maps larger than 256 tiles, this value is scaled up.
STR_CONFIG_SETTING_SNOW_COVERAGE                                :Snow coverage: {STRING2}

> STR_CONFIG_SETTING_SNOW_COVERAGE_HELPTEXT                       :Control how much snow is on the sub-arctic landscape. Snow also affects industry generation and town growth requirements. Only used during map generation and the game tries to approximate the value. There will always be a difference.

Reviewer: ```suggestion
STR_CONFIG_SETTING_SNOW_COVERAGE_HELPTEXT                       :Controls the approximate amount of snow on the sub-arctic landscape. Snow also affects industry generation and town growth requirements. Only used during map generation.
```

STR_CONFIG_SETTING_SNOW_COVERAGE_VALUE                          :{NUM}%
STR_CONFIG_SETTING_DESERT_COVERAGE                              :Desert coverage: {STRING2}

> STR_CONFIG_SETTING_DESERT_COVERAGE_HELPTEXT                     :Control how much desert is on the tropical landscape. Desert also affects industry generation. Only used during map generation and the game tries to approximate the value. There will always be a difference.

Reviewer: ```suggestion
STR_CONFIG_SETTING_DESERT_COVERAGE_HELPTEXT                     :Control the approximate amount of desert on the tropical landscape. Desert also affects industry generation. Only used during map generation.
```

STR_MAPGEN_NUMBER_OF_TOWNS                                      :{BLACK}No. of towns:
STR_MAPGEN_DATE                                                 :{BLACK}Date:
STR_MAPGEN_NUMBER_OF_INDUSTRIES                                 :{BLACK}No. of industries:
STR_MAPGEN_HEIGHTMAP_HEIGHT                                     :{BLACK}Highest mountain:
STR_MAPGEN_HEIGHTMAP_HEIGHT_UP                                  :{BLACK}Increase the maximum height of mountains on the map by one

> STR_MAPGEN_HEIGHTMAP_HEIGHT_DOWN                                :{BLACK}Decrease the maximum height of mountains on the map by one

Reviewer: shouldn't these (and others) have a `_TOOLTIP` suffix?

Author: I think they all do, but given none had, I did not add it. I tried to keep the diff as small as I could, to prevent this PR blowing up any more than it already does.

There are a ton of these inconsistencies around mapgen. For example, sea-level is not in the settings. Sea level query has no own query caption. And indeed, several of these entries are missing `_TOOLTIP` etc.

I rather not fix it for this PR, but if we have some time, this sure could use a good code cleanup :)

STR_MAPGEN_SNOW_COVERAGE                                        :{BLACK}Snow coverage:

> STR_MAPGEN_SNOW_COVERAGE_UP                                     :{BLACK}Increase snow coverage by ten

Reviewer 2: nitpick: shouldn't this be "by then percent" ?

STR_MAPGEN_HEIGHTMAP_SIZE_LABEL                                 :{BLACK}Size:
STR_MAPGEN_HEIGHTMAP_SIZE                                       :{ORANGE}{NUM} x {NUM}

STR_MAPGEN_TERRAIN_TYPE_QUERY_CAPT                              :{WHITE}Target peak height
STR_MAPGEN_HEIGHTMAP_HEIGHT_QUERY_CAPT                          :{WHITE}Change highest mountain height

> STR_MAPGEN_SNOW_COVERAGE_QUERY_CAPT                             :{WHITE}Change snow coverage

Reviewer 2: same as the other ones, but then probably "coverage percentage" perhaps, to better indicate the unit being used.
